course/Program.cs

```diff
@@ -0,0 +1,206 @@
+            var phoneNumber = Console.ReadLine();
+
+            File.AppendAllLines(Filename, new[] { $"{firstName},{lastName},{phoneNumber}" });
+        }
+
+        static void SearchBy(bool byName = false)
+        {
+            Console.Clear();
+            Console.WriteLine("Search condition - {0}", byName ? "if first/last name contains search term" : "if entered number(s) exist in any contact");
+
+            Console.WriteLine("Enter Search Term...");
+            var searchTerm = Console.ReadLine();
+
+            bool found = false;
+            foreach (var (firstName, lastName, phoneNumber) in ReadPhoneBook())
+            {
+                if (byName)
+                {
+                    if (firstName.Contains(searchTerm, StringComparison.OrdinalIgnoreCase) ||
+                        lastName.Contains(searchTerm, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Console.WriteLine($"Found user {firstName} {lastName} with phone {phoneNumber}");
+                        found = true;
+                    }
+                } else
+                {
+                    if (phoneNumber.Contains(searchTerm))
+                    {
+                        Console.WriteLine($"Found user {firstName} {lastName} with phone {phoneNumber}");
+                        found = true;
+                    }
+                }
+            }
+
+            if (!found)
+            {
+                Console.WriteLine("No user found!");
+            }
+
+            Wait();
+        }
+
+        static void ModifyContact()
+        {
+            Console.Clear();
+            var phoneBook = ReadPhoneBook();
+            int index = 0;
+            (string, string, string) userData = ("", "", "");
+            Console.WriteLine("Please, write full the first and the last name of the user");
+
+            Console.WriteLine("Enter Search Term...");
+            var searchTerm = Console.ReadLine();
+
+            bool found = false;
+            foreach (var (firstName, lastName, phoneNumber) in phoneBook)
+            {
+                if ($"{firstName} {lastName}".Equals(searchTerm, StringComparison.OrdinalIgnoreCase))
+                {
+                    userData = ( firstName, lastName, phoneNumber );
+                    found = true;
+                    break;
+                }
+                if (!found)
+                {
+                    index++;
+                }
+            }
+
+            if (index == phoneBook.Length && !found)
+            {
+                Console.WriteLine("No user found!");
+            } else
+            {
+                Console.WriteLine("Please, write new phone number");
+                string newPhoneNumber = Console.ReadLine();
+                userData.Item3 = newPhoneNumber;
+
+                phoneBook[index] = userData;
+
+                string[] convertedPhoneBook = new string[phoneBook.Length+1];
```

Comment:
It's better to extract this stuff to separated method like we did during the lesson, but ok now